### PR TITLE
TBHelper conversion

### LINKED ISSUE #176: Use TBHelper in all C++ test benches
Update all C++ test benches for modules included in the summer chain to use the TBHelper class to access test-vector files.

See slides from 18 March for details on the TBHelper class:
https://indico.cern.ch/event/1010698/#3-generalizing-hls-test-benche
Also, see VMRouter_test.cpp, VMRouterCM_test.cpp, and TrackletCalculator_test.cpp for real-life examples of how to use the class.

### PR DESCRIPTION
**Pull request by aehart**
Resolves #176 for the TrackBuilder. Other modules can push to this PR.

### CODE REVIEW

File: TrackletAlgorithm/MatchCalculator.h
Comment (bryates) [inline TrackletAlgorithm/MatchCalculator.h:837]: This is a some what complicated solution I came up with to count how many slots are not masked. That way, seeding in L5L6 (which would be slot 3) is placed in the correct slot (slot 2 if the TBHelper is only reading L1L2 and L5L6). This relies on the mask to agree with the number of files TBHelper loads, which seems reasonable.

Comment (aehart) [reply TrackletAlgorithm/MatchCalculator.h:838]: @bryates I'm not sure I understand what's happening here. I thought the `FMMask` essentially told you which FM memories are "on" for a given MC. That way, the size of the `fullmatch` array can be the same for every MC, but HLS will only generate ports for the memories that are enabled by the mask.

So why would the index of, e.g., the L5L6 FM memory ever change?

Comment (bryates) [reply TrackletAlgorithm/MatchCalculator.h:838]: `FMMask` disables ports so Vivado will trim them, but this does not change the index, which was hard coded to 3 in the past. `FMCount` allows it to determine how many ports are actually active, and allocate the correct spot. I needed this because the `TBHelper` uses vectors that are only as large as the active ports.

**Comment by aehart** [comment]
FYI, I just rebased and removed the commits related to the MatchProcessor, so as to expedite the review for modules in the summer chain.

@bryates I moved the MP commits to the mp_tb_helper branch. Please wait until this PR is merged to open a PR for that branch.

**Review by pwittich — APPROVED** [review]

**Pull request merged**
